File: sdks/dotnet/src/Models/Customer.cs
using System;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class Customer
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("email")]
        public string Email { get; set; }

        [JsonProperty("phone")]
        public string Phone { get; set; }

        [JsonProperty("address")]
        public string Address { get; set; }

        [JsonProperty("customer_type")]
        public string CustomerType { get; set; }

        [JsonProperty("loyalty_points")]
        public int LoyaltyPoints { get; set; }

        [JsonProperty("store_credit_balance")]
        public double StoreCreditBalance { get; set; }

        [JsonProperty("total_spend")]
        public double TotalSpend { get; set; }

        [JsonProperty("tier")]
        public string Tier { get; set; }

        [JsonProperty("tier_name")]
        public string TierName { get; set; }

        [JsonProperty("marketing_opt_in")]
        public bool MarketingOptIn { get; set; }

        [JsonProperty("created_at")]
        public DateTime? CreatedAt { get; set; }
    }
}

File: sdks/php/src/Models/Customer.php
<?php

namespace Puxbay\Models;

class Customer
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
        public readonly ?string $email = null,
        public readonly ?string $phone = null,
        public readonly ?string $address = null,
        public readonly string $customerType,
        public readonly int $loyaltyPoints,
        public readonly float $storeCreditBalance,
        public readonly float $totalSpend,
        public readonly ?string $tier = null,
        public readonly ?string $tierName = null,
        public readonly bool $marketingOptIn,
        public readonly ?array $metadata = null,
        public readonly \DateTimeImmutable $createdAt,
    ) {
    }

    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            name: $data['name'],
            email: $data['email'] ?? null,
            phone: $data['phone'] ?? null,
            address: $data['address'] ?? null,
            customerType: $data['customer_type'],
            loyaltyPoints: (int) $data['loyalty_points'],
            storeCreditBalance: (float) $data['store_credit_balance'],
            totalSpend: (float) $data['total_spend'],
            tier: $data['tier'] ?? null,
            tierName: $data['tier_name'] ?? null,
            marketingOptIn: $data['marketing_opt_in'] ?? false,
            metadata: $data['metadata'] ?? null,
            createdAt: new \DateTimeImmutable($data['created_at']),
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'email' => $this->email,
            'phone' => $this->phone,
            'address' => $this->address,
            'customer_type' => $this->customerType,
            'loyalty_points' => $this->loyaltyPoints,
            'store_credit_balance' => $this->storeCreditBalance,
            'marketing_opt_in' => $this->marketingOptIn,
            'metadata' => $this->metadata,
        ];
    }
}
